src/ingress.rs

```diff
@@ -144,6 +145,43 @@ where
         packet_tx
     }
 
+    pub fn init_packet_processor_with_channel(
+        &mut self,
+        packet_tx: UnboundedSender<In>,
+        mut packet_rx: UnboundedReceiver<In>,
+    ) {
+        let mut join_set = JoinSet::new();
+        let router = self.router.clone();
+        let signal_queue = self.signal_queue.clone();
+        let _nw_handle = join_set.spawn(
+            async move {
+                debug!(target: "state_machine", spawning = "IngressAdapter.packet_tx");
+                while let Some(packet) = packet_rx.recv().await {
+                    trace!("receiving packet");
+                    let id = match router.get_id(&packet) {
+                        Err(e) => {
+                            error!(err = ?e, ?packet, "could not get id from router");
+                            continue;
+                        }
+                        Ok(None) => {
+                            warn!(?packet, "unable to route packet");
+                            continue;
+                        }
+                        Ok(Some(state_id)) => state_id,
+                    };
+                    K::Input::try_from(packet)
+                        .map(|input| {
+                            signal_queue.push_back(Signal { id, input });
+                        })
+                        .log_attached_err("ia::processors from packet failed");
+                }
+            }
```

Comment:
could we encapsulate this in a method that returns `impl Future<Output = ()>` since this block is identical to the one in `init_packet_processor_with_handle`?

Reply:
I'll see what I can do